compiler/resolution/interfaceResolution.cpp

```diff
@@ -984,14 +986,165 @@ static void removeCallEtc(BlockStmt* holder, CallExpr* call,
   call->remove();
 }
 
-static bool checkNonemptyHolder(InterfaceSymbol* isym,  ImplementsStmt* istm,
-                                BlockStmt*     holder,  FnSymbol*     target,
-                                FnSymbol*       reqFn,  bool  reportErrors) {
+//
+// This looks at in-intent formals for checkAdjustCall().
+// For an in-intent formal, a chpl__initCopy() may get introduced
+// to copy the original actual of 'call':
+//
+//   def _formal_tmp_
+//   move _formal_tmp_, chpl__initCopy(origActual)
+//   // 'call' now contains _formal_tmp_ in lieu of origActual
+//
+// This copy is not needed and will be elided later in elideCopies().
+// It should not necessitate a wrapper.
+// We still remove this copy here to leave 'holder' empty - consistently
+// with our determination that the wrapper is not needed.
+// This copy will be reintroduced upon 'resolveFunction(wrapper)' - if we
+// end up creating the wrapper because of other arguments - and elided later.
+//
+// Returns true if this is the only conversion for the given argument
+// i.e. the wrapper is not required because of this argument.
+//
+static bool removedInitCopyForInArg(SymExpr* actualSE, Symbol* actualSym,
+                                    ArgSymbol* origActual) {
+  if (! (origActual->intent & INTENT_FLAG_IN))
+    return false;  // not 'in' intent
+  
+  if (actualSym->type != origActual->type)
+    return false;  // if the types differ, we probably must keep the copy
+
+  // Find the copying pattern in the AST
+  if (SymExpr* actualDef = actualSym->getSingleDef())
+   if (CallExpr* move = toCallExpr(actualDef->parentExpr))
+    if (CallExpr* src = toCallExpr(move->get(2)))
+     if (src->isNamed("chpl__initCopy"))
+      if (SymExpr* icSrc = toSymExpr(src->get(1)))
+       // Report success only if initCopy is the only call here.
+       // Otherwise preserve everything in the wrapper.
+       if (icSrc->symbol() == origActual) {
+         // Switch 'call' to use 'icSrc' instead of 'actualSym'.
+         // Remove 'actualSym' altogether.
+         actualSE->replace(icSrc->remove());
+         move->remove();
+         actualSym->defPoint->remove();
+         // 'actualSym' is expected to be a temp and have no other refs.
+         INT_ASSERT(actualSym->firstSymExpr() == nullptr);
+         return true;
+       }
+
+  return false;  // no changes were made; need a wrapper
+}
+
+//
+// We use this test to decide whether the target function needs any
+// conversions from the actuals (in 'formalDups') - whose types come
+// from the req.fn / implements statement we are resolving - to the formals
+// of the target function. If we do, we will wrap these conversions, see
+// finalizeHolder(). tryResolveCall(call) inserts such conversions into
+// the holder block if necessary. Given that the holder may contain
```

Comment:
"Given that the holder" needs a comma - e.g. "Given that, the holder"; or else more to the sentence e.g. "Given that the holder ... we need to ..."